### ISSUE TITLE
inner1d sometimes dumps core on arrays whose shape contains 0 (migrated from Trac #1956)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3504]
Original ticket http://projects.scipy.org/numpy/ticket/1956
Reported 2011-09-29 by trac user tbarnetlamb, assigned to unknown.

The following appears to be an error:

In [1]: import numpy.version

In [2]: numpy.version.version
Out[2]: '1.4.1'

In [3]: from numpy.core.umath_tests import inner1d

In [4]: inner1d(numpy.zeros((0,)), numpy.zeros((0,))) # okay
Out[4]: 0.0

In [5]: inner1d(numpy.zeros((0, 2)), numpy.zeros((0, 2))) # crashes
Fatal Python error: PyEval_RestoreThread: NULL tstate
Abort

**Comment by thouis** [comment]
Comment in Trac by trac user tbarnetlamb, 2011-09-29

Better formatting:

```
    In [1]: import numpy.version

    In [2]: numpy.version.version
    Out[2]: '1.4.1'

    In [3]: from numpy.core.umath_tests import inner1d

    In [4]: inner1d(numpy.zeros((0,)), numpy.zeros((0,))) # okay
    Out[4]: 0.0

    In [5]: inner1d(numpy.zeros((0, 2)), numpy.zeros((0, 2))) # crashes
    Fatal Python error: PyEval_RestoreThread: NULL tstate
    Abort
```

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2011-09-29

Raises a ValueError in development branch:

```
In [4]: inner1d(numpy.zeros((0, 2)), numpy.zeros((0, 2))) 
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/home/charris/<ipython-input-4-6e74a21a5507> in <module>()
----> 1 inner1d(numpy.zeros((0, 2)), numpy.zeros((0, 2)))

ValueError: Iteration of zero-sized operands is not enabled
```

It appears that this bug is fixed in version 1.6.1, if not earlier.
